accounts/abi/bind/base.go

```diff
@@ -494,8 +494,10 @@ func (c *BoundContract) WatchLogs(opts *WatchOpts, name string, query ...[]inter
 
 // UnpackLog unpacks a retrieved log into the provided output structure.
 func (c *BoundContract) UnpackLog(out interface{}, event string, log types.Log) error {
-	if log.Topics[0] != c.abi.Events[event].ID {
-		return fmt.Errorf("event signature mismatch")
+	if len(log.Topics) > 0 {
```

Comment:
Hmm it looks like there is already an additional assumption here that the `event` is actually part of the ABI since otherwise there will be a nil pointer error when accessing the ID as well as when accessing the inputs on line 508.

I think a better change would be to clarify what the exact invariants are here and document them.

Reply:
hshould we return an error in case `len(log.topics) < 1`?